hail/src/test/scala/is/hail/annotations/UnsafeSuite.scala

```diff
@@ -349,25 +349,21 @@ class UnsafeSuite extends HailSuite {
   @Test def testRegionAllocation() {
     val pool = RegionPool.get
 
-    case class Counts(regions: Int, freeRegions: Int, freeBlocks: Int) {
+    case class Counts(regions: Int, freeRegions: Int) {
```

Comment:
This test is breaking on this PR, so I've refactored it slightly by taking out the free block tracking. Now that there are separate block pools for blocks of different sizes, the behavior of these test w.r.t. the number of free blocks is dependent on whether or not tests with smaller blockSizes have been run previously on this thread, which is undesirable and unpredictable. 

The block behavior is already tested elsewhere. This test was meant to check that region dependencies allowed regions to be freed in the correct places, which is still being tested here.